### PR TITLE
Ensure Template can render from a subclass of String

### PR DESCRIPTION
**Pull request by rhymes**
Currently `Template.parse().render` fails if the inputed string is not a `String` object but a subclass of `String` , eg. `ActiveSupport::SafeBuffer` used in Rails apps.

By switching `instance_of?()` to `is_a?()` when checking a node, we can avoid this.

See https://github.com/Shopify/liquid/issues/1390 for further details.

### CODE REVIEW

**Comment by rhymes** [comment]
FYI: I did sign the CLA after the notice, pleaes re-run the CLA check

**Review by pushrax — APPROVED** [review]

**Comment by pushrax** [comment]
Thank you for the contribution. LGTM

For reasons unknown to me, re-running the CLA check isn't updating the status. Might be cached by commit. You could try amending the commit to force all the checks to run again.

**Comment by rhymes** [comment]
@pushrax done, thank you!

**Comment by pushrax** [comment]
Although I originally didn't think it would be required, I suppose we also need to update `BlockBody#remove_blank_strings` since Liquid may generate strings into the nodelist by splitting the input source. I believe splitting a SafeBuffer will produce more SafeBuffers. Do you want to tackle that in this PR as well? If not we can do it in a followup.

**Comment by rhymes** [comment]
@pushrax makes sense, I'm going to add it to this PR

**Comment by pushrax** [comment]
cc @dylanahsmith I should ask, was the strict requirement on String intentional for Liquid-C purposes? I assume this change is fine because Check_Type will raise before it ever becomes an issue https://github.com/Shopify/liquid-c/blob/f24b503024c5ffd4f200d1490ea3549a340adabd/ext/liquid_c/tokenizer.c#L50

File: lib/liquid/block_body.rb [hunk 1]
```diff
@@ -193,7 +193,7 @@ def blank?
     # Note that it is now preferred to use the `liquid` tag for this use case.
     def remove_blank_strings
       raise "remove_blank_strings only support being called on a blank block body" unless @blank
-      @nodelist.reject! { |node| node.instance_of?(String) }
+      @nodelist.reject! { |node| node.is_a?(String) }
```
Comment (dylanahsmith) [inline lib/liquid/block_body.rb:196]: Do you actually have a use case for string subclass being used in all these derived string classes?  Can't we just coerce to `String` before tokenizing?

Comment (pushrax) [reply lib/liquid/block_body.rb:196]: Also seems like a viable solution. The author's use case is due to use of ActiveSupport::SafeBuffer in the application.

Do you think that should go in `Tokenizer#initialize`, and duplicated in `C::Tokenizer#initialize`?

Comment (dylanahsmith) [reply lib/liquid/block_body.rb:196]: Yeah.  We can just use `source = source.to_str unless source.instance_of?(String)` for the coercion.

Comment (dylanahsmith) [reply lib/liquid/block_body.rb:196]: Doesn't look like this is a problem for liquid-c

**Pull request closed without merging**